gift-certificate-application/controller/src/main/java/com/epam/esm/controller/GiftCertificateController.java

```diff
@@ -0,0 +1,74 @@
+package com.epam.esm.controller;
+
+import com.epam.esm.controller.hateoas.LinkBuilder;
+import com.epam.esm.pagination.CustomPagination;
+import com.epam.esm.entity.GiftCertificate;
+import com.epam.esm.service.GiftCertificateService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
+@RestController
+@RequestMapping(value = "/gift-certificates")
+public class GiftCertificateController {
+    private final GiftCertificateService giftCertificateService;
+    private final LinkBuilder<GiftCertificate> giftCertificateLinkBuilder;
+
+    @Autowired
+    public GiftCertificateController(GiftCertificateService giftCertificateService, LinkBuilder<GiftCertificate> giftCertificateLinkBuilder) {
+        this.giftCertificateService = giftCertificateService;
+        this.giftCertificateLinkBuilder = giftCertificateLinkBuilder;
+    }
+
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public GiftCertificate create(@RequestBody (required = false) GiftCertificate giftCertificate) {
```

Comment:
Remove space between @RequestBode and bracket
Also, why is giftCertificate not required?

Reply:
I removed (required = false)